### PR TITLE
JSON or as a YAML file: workflow.states must be int32

### PR DESCRIPTION
**Pull request by spolti**
Fixes https://github.com/serverlessworkflow/sdk-go/issues/177

**Many thanks for submitting your Pull Request :heart:!**

**What this PR does / why we need it**:

**Special notes for reviewers**:

**Additional information (if needed):**

### CODE REVIEW

File: parser/testdata/workflows/greetings-v08-spec.sw.yaml [hunk 1]
```diff
@@ -170,6 +170,7 @@ states:
     type: inject
     data:
       result: Hello World, last state!
+      boolValue: false
```
Comment (ricardozanini) [inline parser/testdata/workflows/greetings-v08-spec.sw.yaml:173]: Can you also add the test case where originated this problem?

**Comment by ricardozanini** [comment]
@ribeiromiranda can you also take a look, please?

File: parser/testdata/workflows/greetings-v08-spec.sw.yaml (hunk 1 above)
Comment (spolti) [reply parser/testdata/workflows/greetings-v08-spec.sw.yaml:173]: Not sure if I follow, this boolean example has caused the issue.
What else would you like to see in the new test case?

Comment (ricardozanini) [reply parser/testdata/workflows/greetings-v08-spec.sw.yaml:173]: The SW that originated this problem. Wasn't the `compensate: true`, the problem?

Comment (spolti) [reply parser/testdata/workflows/greetings-v08-spec.sw.yaml:173]: No, only this field, as the object struct was dealing only with string, raw or int, and int was the default, thus we had the int32 error in the message.

the states transition.compensate is correctly being marshaled/unmarshaled.

but I see your point, will add a test case with the entire flow.

**Review by ricardozanini — APPROVED** [review]

**Comment by spolti** [comment]
can we merge this one?

**Pull request merged**